dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,988 @@
+# -*- coding: utf-8 -*-
+"""Generates a GCP cloud resource tree."""
+
+import json
+import tempfile
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+from typing import Dict, List, Optional, Any
+
+from libcloudforensics.providers.gcp.internal.common import CreateService
+from libcloudforensics.providers.gcp.internal.log import GoogleCloudLog
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+from dftimewolf.lib.processors.gcp_cloud_resource_tree_helper import OperatingMode, Resource  # pylint: disable=line-too-long
+
+
+class GCPCloudResourceTree(module.BaseModule):
+  """GCP Cloud Resource Tree Creator.
+
+  Attributes:
+    project_id (str): Id of the project where the resource is located.
+    resource_name (str): Name of the resource to build the tree for.
+    resource_type (str): Resource type.
+  """
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str] = None,
+               critical: bool = True) -> None:
+    """Initializes the Cloud Resource Tree Processor.
+
+    Args:
+      state: recipe state.
+      name: The module's runtime name.
+      critical: True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(GCPCloudResourceTree, self).__init__(state,
+                                               name=name,
+                                               critical=critical)
+
+    self.project_id: str = ''
+    self.resource_name: str = ''
+    self.resource_type: str = ''
+    self.mode: OperatingMode = OperatingMode.ONLINE
+    self.period_covered_by_retrieved_logs: Dict[str, datetime] = {}
+    self.resources_dict: Dict[str, Resource] = {}
+
+  # pylint: disable=arguments-differ
+  def SetUp(self, project_id: str, resource_name: str, resource_type: str,
+            mode: str) -> None:
+    """Sets up the resource we want to build the tree for.
+
+    Args:
+      project_id: Project id where the resources are located.
+      resource_name: Resource name.
```

Comment:
is there a reason why not to have resource id as a parameter as well and make sure the user provide either resource id or resource name is available? sometimes there might be two disks or instances with the same name because one was deleted and then the next was created, in this case the id is the way to differentiate between these two instances, also when automation take-care of starting and shutting down resources (ex: managed groups) this case will happen often

Reply:
Added the ability to supply resource_id